### PR TITLE
Add `onError` handler to useAssistant hook

### PR DESCRIPTION
**Pull request by mfts**
similar to `useChat` exposing an `onError` handler on the useAssistant hook. 

```tsx
const { status, messages, input, submitMessage, handleInputChange, error } =
    useAssistant({
      api: '/api/assistant',
      onError: err => { 
        console.error(err.message);
      },
    });
```

### CODE REVIEW

**Review by lgrammel — COMMENTED** [review]
Thanks, this looks great!

File: packages/core/react/use-assistant.ts [hunk 1]
```diff
@@ -53,12 +54,13 @@ export function experimental_useAssistant({
...
   headers,
   body,
+  onError,
 }: UseAssistantOptions): UseAssistantHelpers {
   const [messages, setMessages] = useState<Message[]>([]);
   const [input, setInput] = useState('');
   const [threadId, setThreadId] = useState<string | undefined>(undefined);
   const [status, setStatus] = useState<AssistantStatus>('awaiting_message');
-  const [error, setError] = useState<unknown | undefined>(undefined);
+  const [error, setError] = useState<unknown | Error>(undefined);
```
[the remark this answers is not in the document]
Comment (mfts) [reply packages/core/react/use-assistant.ts]: Thanks for picking that up.

**Review by lgrammel — APPROVED** [review]

**Comment by lgrammel** [comment]
Build failure seems to be related to changes in PR.

**Comment by mfts** [comment]
> Build failure seems to be related to changes in PR.

Sorry for the slow response. I created a proper `Error` and passed it in. That should fix the issue.

**Comment by mfts** [comment]
fixed the prettier format error

**Review by MaxLeiter — CHANGES_REQUESTED** [review]
Thanks @mfts! Can you update the `OptionTable` in `docs/pages/docs/api-reference/use-assistant.mdx`?

**Comment by mfts** [comment]
Added documentation

**Review by MaxLeiter — APPROVED** [review]

**Pull request merged**